src/serac/physics/heat_transfer.hpp

```diff
@@ -649,34 +655,103 @@ class HeatTransfer<order, dim, Parameters<parameter_space...>, std::integer_sequ
+                                    *parameters_[parameter_indices].state...);
+      auto jacobian  = assemble(drdu);
+      auto J_T       = std::unique_ptr<mfem::HypreParMatrix>(jacobian->Transpose());
+
+      for (const auto& bc : bcs_.essentials()) {
+        bc.apply(*J_T, temperature_adjoint_load_vector, adjoint_essential);
+      }
+
+      lin_solver.SetOperator(*J_T);
+      lin_solver.Mult(temperature_adjoint_load_vector, adjoint_temperature_);
+
+      // Reset the equation solver to use the full nonlinear residual operator
+      nonlin_solver_->setOperator(residual_with_bcs_);
+
+      return {{"adjoint_temperature", adjoint_temperature_}};
+    }
+
+    SLIC_ERROR_ROOT_IF(ode_.GetTimestepper() != TimestepMethod::BackwardEuler,
+                       "Only backward Euler implemented for transient adjoint heat conduction.");
+
+    SLIC_ERROR_ROOT_IF(cycle_ == 0,
+                       "Maximum number of adjoint timesteps exceeded! The number of adjoint timesteps must equal the "
+                       "number of forward timesteps");
+
+    // Load the temperature from the previous cycle from disk
+    serac::FiniteElementState temperature_n_minus_1(temperature_);
+    StateManager::loadPreviousStates(cycle_, {temperature_});
+    StateManager::loadPreviousStates(cycle_ - 1, {temperature_n_minus_1});
+
+    temperature_rate_ = temperature_;
+    temperature_rate_.Add(-1.0, temperature_n_minus_1);
+    temperature_rate_ /= dt_;
+
+    // K := dR/du
+    auto K = serac::get<DERIVATIVE>((*residual_)(differentiate_wrt(temperature_), temperature_rate_,
+                                                 shape_displacement_, *parameters_[parameter_indices].state...));
+    std::unique_ptr<mfem::HypreParMatrix> k_mat(assemble(K));
+
+    // M := dR/du_dot
+    auto M = serac::get<DERIVATIVE>((*residual_)(temperature_, differentiate_wrt(temperature_rate_),
+                                                 shape_displacement_, *parameters_[parameter_indices].state...));
+    std::unique_ptr<mfem::HypreParMatrix> m_mat(assemble(M));
+
+    J_.reset(mfem::Add(1.0, *m_mat, dt_, *k_mat));
+    auto J_T = std::unique_ptr<mfem::HypreParMatrix>(J_->Transpose());
+
+    // recall that temperature_adjoint_load_vector and d_temperature_dt_adjoint_load_vector were already multiplied by
+    // -1 above
+    mfem::HypreParVector modified_RHS(temperature_adjoint_load_vector);
+    modified_RHS *= dt_;
+    modified_RHS.Add(1.0, temperature_rate_adjoint_load_vector);
+    modified_RHS.Add(-dt_, _d_temperature_start_of_step_);
+
     for (const auto& bc : bcs_.essentials()) {
-      bc.apply(*J_T, adjoint_load_vector, adjoint_essential);
+      bc.apply(*J_T, modified_RHS, adjoint_essential);
     }
 
     lin_solver.SetOperator(*J_T);
-    lin_solver.Mult(adjoint_load_vector, adjoint_temperature_);
+    lin_solver.Mult(modified_RHS, adjoint_temperature_);
+
+    m_mat->Mult(adjoint_temperature_, _d_temperature_start_of_step_);
```

Comment:
I assume this was the term I was missing? I haven't dug out my notes yet...

Reply:
yeah, I kind of re-derived things, so I'm not exactly sure how things match up with what was before.  This extra implicit sensitivity field certainly helps me reason through the math easier.  I think I'll chuck in a ~4 line comment for the derivation to set notation, and so maybe future people can figure what in the world I'm up to here.

Reply:
This looks right to me. I think we could re-factor later to avoid some of the matvecs, but that's an optimization for the future.